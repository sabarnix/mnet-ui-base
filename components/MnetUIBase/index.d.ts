import * as React from "react";
import { ThemeType } from "../../themes";
import { BackgroundType } from '../../utils';

export interface MnetUIBaseProps {
  background?: BackgroundType;
  cssVars?: boolean;
  dir?: "rtl";
  full?: boolean;
  plain?: boolean;
  theme?: ThemeType;
  themeMode?: "dark" | "light";
  userAgent?: string;
}

declare const MnetUIBase: React.ComponentClass<MnetUIBaseProps & JSX.IntrinsicElements['div']>;

export { MnetUIBase };
