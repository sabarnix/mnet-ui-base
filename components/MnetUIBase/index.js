"use strict";

exports.__esModule = true;
exports.MnetUIBase = void 0;

var _MnetUIBase = require("./MnetUIBase");

exports.MnetUIBase = _MnetUIBase.MnetUIBase;