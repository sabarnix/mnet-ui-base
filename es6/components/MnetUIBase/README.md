## MnetUIBase
The top level MnetUIBase container.

[![](https://cdn-images-1.medium.com/fit/c/120/120/1*TD1P0HtIH9zF0UEH28zYtw.png)](https://storybook.grommet.io/?selectedKind=MnetUIBase&full=0&addons=0&stories=1&panelRight=0) [![](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/grommet/grommet-sandbox?initialpath=mnetuibase&module=%2Fsrc%2FMnetUIBase.js)
## Usage

```javascript
import { MnetUIBase } from 'mnet-ui-base';
<MnetUIBase>...</MnetUIBase>
```

## Properties

**background**

Either a color 
identifier to use for the background color. For example: 'neutral-1'. Or, a 
'url()' for an image. Dark is not needed if color is provided.

```
string
{
  color: 
    string
    {
      dark: string,
      light: string
    },
  dark: 
    boolean
    string,
  image: string,
  position: string,
  opacity: 
    string
    boolean
    number
    weak
    medium
    strong,
  repeat: 
    no-repeat
    repeat
    string,
  size: 
    cover
    contain
    string,
  light: string
}
```

**dir**

Layout direction for right to left contexts

```
rtl
```

**full**

Whether to take the whole viewport.

```
boolean
```

**plain**

Whether or not MnetUIBase should apply a global font-family, font-size,
        and line-height.

```
boolean
```

**cssVars**

Whether to expose the css variables.

```
boolean
```

**theme**

Custom styles for MnetUIBase app component.

```
object
```

**themeMode**

Dark vs. light theme variation. Default is unspecified and left to
      theme.

```
dark
light
```

**userAgent**

User agent used to detect the device width for setting the initial
      breakpoint.

```
string
```
  
## Intrinsic element

```
div
```
## Theme
  
**mnet.extend**

Any additional style for MnetUIBase. Expects `string | (props) => {}`.

Defaults to

```
undefined
```

**global.font.face**

Custom font face declaration Expects `string | (props) => {}`.

Defaults to

```
undefined
```
