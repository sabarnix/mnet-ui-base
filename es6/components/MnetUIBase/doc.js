import { describe, PropTypes } from 'react-desc';
import { backgroundDoc, getAvailableAtBadge } from '../../utils';
export var doc = function doc(MnetUIBase) {
  var DocumentedMnetUIBase = describe(MnetUIBase).availableAt(getAvailableAtBadge('MnetUIBase')).description('The top level MnetUIBase container.').usage("import { MnetUIBase } from 'mnet-ui-base';\n<MnetUIBase>...</MnetUIBase>").intrinsicElement('div');
  DocumentedMnetUIBase.propTypes = {
    background: backgroundDoc,
    dir: PropTypes.oneOf(['rtl']).description('Layout direction for right to left contexts'),
    full: PropTypes.bool.description('Whether to take the whole viewport.').defaultValue(false),
    plain: PropTypes.bool.description("Whether or not MnetUIBase should apply a global font-family, font-size,\n        and line-height.").defaultValue(false),
    cssVars: PropTypes.bool.description('Whether to expose the css variables.').defaultValue(false),
    theme: PropTypes.object.description('Custom styles for MnetUIBase app component.'),
    themeMode: PropTypes.oneOf(['dark', 'light']).description("Dark vs. light theme variation. Default is unspecified and left to\n      theme."),
    userAgent: PropTypes.string.description("User agent used to detect the device width for setting the initial\n      breakpoint.")
  };
  return DocumentedMnetUIBase;
};
export var themeDoc = {
  'mnet.extend': {
    description: 'Any additional style for MnetUIBase.',
    type: 'string | (props) => {}',
    defaultValue: undefined
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined
  }
};