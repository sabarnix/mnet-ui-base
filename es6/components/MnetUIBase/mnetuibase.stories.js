import React from 'react';
import { storiesOf } from '@storybook/react';
import { mnet, MnetUIBase, Anchor, Box } from 'mnet-ui-base';
import { Add } from "grommet-icons/es6/icons/Add";
var customTheme = {
  global: {
    colors: {
      custom: '#cc6633'
    }
  }
};

var Themed = function Themed() {
  return /*#__PURE__*/React.createElement(MnetUIBase, {
    theme: customTheme
  }, /*#__PURE__*/React.createElement(Box, {
    pad: "medium"
  }, /*#__PURE__*/React.createElement(Anchor, {
    icon: /*#__PURE__*/React.createElement(Add, null),
    label: "Add",
    color: "custom"
  })));
};

var Plain = function Plain() {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(MnetUIBase, {
    plain: true
  }, /*#__PURE__*/React.createElement(Box, {
    pad: "medium"
  }, /*#__PURE__*/React.createElement("p", null, "Plain MnetUIBase"))), /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Box, {
    pad: "medium"
  }, /*#__PURE__*/React.createElement("p", null, "Not plain MnetUIBase"))));
};

var MnetUIBaseVars = function MnetUIBaseVars() {
  return /*#__PURE__*/React.createElement(MnetUIBase, {
    theme: mnet,
    cssVars: true
  }, /*#__PURE__*/React.createElement(Box, {
    pad: "medium",
    background: "var(--accent-2)",
    gap: "medium"
  }, /*#__PURE__*/React.createElement(Box, null, "Checkout MnetUIBase variables, you can find them in the StyledMnetUIBase DOM."), /*#__PURE__*/React.createElement(Box, {
    "with": true
  }, "For example, the background color in this Box is using var(--accent-2)")));
};

storiesOf('MnetUIBase', module).add('Plain', function () {
  return /*#__PURE__*/React.createElement(Plain, null);
}).add('Theme', function () {
  return /*#__PURE__*/React.createElement(Themed, null);
}).add('Vars', function () {
  return /*#__PURE__*/React.createElement(MnetUIBaseVars, null);
});