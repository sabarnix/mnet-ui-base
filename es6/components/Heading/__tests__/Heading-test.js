import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { MnetUIBase } from '../../MnetUIBase';
import { Heading } from '..';
test('Heading renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Heading, null)));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Heading level renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Heading, {
    level: 1
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 2
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 3
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 4
  }), /*#__PURE__*/React.createElement(Heading, {
    level: "1"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: "2"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: "3"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: "4"
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Heading size renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Heading, {
    level: 1,
    size: "small"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 1,
    size: "medium"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 1,
    size: "large"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 1,
    size: "xlarge"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 2,
    size: "small"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 2,
    size: "medium"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 2,
    size: "large"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 2,
    size: "xlarge"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 3,
    size: "small"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 3,
    size: "medium"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 3,
    size: "large"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 3,
    size: "xlarge"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 4,
    size: "small"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 4,
    size: "medium"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 4,
    size: "large"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 4,
    size: "xlarge"
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 1,
    size: "77px"
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Heading textAlign renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Heading, {
    textAlign: "start"
  }), /*#__PURE__*/React.createElement(Heading, {
    textAlign: "center"
  }), /*#__PURE__*/React.createElement(Heading, {
    textAlign: "end"
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Heading margin renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Heading, {
    margin: "small"
  }), /*#__PURE__*/React.createElement(Heading, {
    margin: "medium"
  }), /*#__PURE__*/React.createElement(Heading, {
    margin: "large"
  }), /*#__PURE__*/React.createElement(Heading, {
    margin: "none"
  }), /*#__PURE__*/React.createElement(Heading, {
    margin: {
      bottom: 'small'
    }
  }), /*#__PURE__*/React.createElement(Heading, {
    margin: {
      top: 'small'
    }
  }), /*#__PURE__*/React.createElement(Heading, {
    margin: {
      bottom: 'none'
    }
  }), /*#__PURE__*/React.createElement(Heading, {
    margin: {
      top: 'none'
    }
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Heading color renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Heading, {
    color: "brand"
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
var LONG = 'a b c d e f g h i j k l m n o p q r s t u v w x y z';
test('Heading truncate renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Heading, {
    truncate: false
  }, LONG), /*#__PURE__*/React.createElement(Heading, {
    truncate: true
  }, LONG)));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('responsive renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Heading, {
    responsive: true
  }), /*#__PURE__*/React.createElement(Heading, {
    responsive: false
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Theme based font family renders', function () {
  var customTheme = {
    heading: {
      font: {
        family: 'Fira Sans'
      },
      level: {
        1: {
          font: {
            family: 'Arial'
          }
        },
        2: {
          font: {
            family: 'Roboto'
          }
        },
        3: {
          font: {
            family: 'Ubuntu'
          }
        }
      }
    }
  };
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, {
    theme: customTheme
  }, /*#__PURE__*/React.createElement(Heading, {
    level: 1
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 2
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 3
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 4
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Theme based font weight renders', function () {
  var customTheme = {
    heading: {
      weight: 600,
      level: {
        1: {
          font: {
            weight: '700'
          }
        },
        2: {
          font: {
            weight: '400'
          }
        },
        3: {
          font: {
            weight: '200'
          }
        }
      }
    }
  };
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, {
    theme: customTheme
  }, /*#__PURE__*/React.createElement(Heading, {
    level: 1
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 2
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 3
  }), /*#__PURE__*/React.createElement(Heading, {
    level: 4
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});