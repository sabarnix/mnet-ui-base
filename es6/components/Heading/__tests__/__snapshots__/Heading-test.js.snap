// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading color renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
  color: #E15151;
}

@media only screen and (max-width:512px) {
  .c1 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
</div>
`;

exports[`Heading level renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

.c2 {
  font-size: 22.666666666666664px;
  line-height: 26.666666666666664px;
  max-width: 362.66666666666663px;
  font-weight: 600;
}

.c3 {
  font-size: 17.333333333333332px;
  line-height: 21.333333333333332px;
  max-width: 277.3333333333333px;
  font-weight: 600;
}

.c4 {
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
  font-weight: 600;
}

@media only screen and (max-width:512px) {
  .c1 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c2 {
    font-size: 17.333333333333332px;
    line-height: 21.333333333333332px;
    max-width: 277.3333333333333px;
  }
}

@media only screen and (max-width:512px) {
  .c3 {
    font-size: 12px;
    line-height: 16px;
    max-width: 192px;
  }
}

@media only screen and (max-width:512px) {
  .c4 {
    font-size: 12px;
    line-height: 16px;
    max-width: 192px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <h2
    className="c2"
  />
  <h3
    className="c3"
  />
  <h4
    className="c4"
  />
  <h1
    className="c1"
  />
  <h2
    className="c2"
  />
  <h3
    className="c3"
  />
  <h4
    className="c4"
  />
</div>
`;

exports[`Heading margin renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  margin: 4px;
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

.c2 {
  margin: 8px;
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

.c3 {
  margin: 16px;
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

.c4 {
  margin: 0px;
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

.c5 {
  margin-bottom: 4px;
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

.c6 {
  margin-top: 4px;
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

.c7 {
  margin-bottom: 0px;
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

.c8 {
  margin-top: 0px;
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

@media only screen and (max-width:512px) {
  .c1 {
    margin: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c1 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c2 {
    margin: 8px;
  }
}

@media only screen and (max-width:512px) {
  .c2 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c3 {
    margin: 16px;
  }
}

@media only screen and (max-width:512px) {
  .c3 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c4 {
    margin: 0px;
  }
}

@media only screen and (max-width:512px) {
  .c4 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c5 {
    margin-bottom: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c5 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c6 {
    margin-top: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c6 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c7 {
    margin-bottom: 0px;
  }
}

@media only screen and (max-width:512px) {
  .c7 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c8 {
    margin-top: 0px;
  }
}

@media only screen and (max-width:512px) {
  .c8 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <h1
    className="c2"
  />
  <h1
    className="c3"
  />
  <h1
    className="c4"
  />
  <h1
    className="c5"
  />
  <h1
    className="c6"
  />
  <h1
    className="c7"
  />
  <h1
    className="c8"
  />
</div>
`;

exports[`Heading renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

@media only screen and (max-width:512px) {
  .c1 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
</div>
`;

exports[`Heading size renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 22.666666666666664px;
  line-height: 26.666666666666664px;
  max-width: 362.66666666666663px;
  font-weight: 600;
}

.c2 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

.c3 {
  font-size: 54.666666666666664px;
  line-height: 58.666666666666664px;
  max-width: 874.6666666666666px;
  font-weight: 600;
}

.c4 {
  font-size: 76px;
  line-height: 80px;
  max-width: 1216px;
  font-weight: 600;
}

.c5 {
  font-size: 17.333333333333332px;
  line-height: 21.333333333333332px;
  max-width: 277.3333333333333px;
  font-weight: 600;
}

.c6 {
  font-size: 44px;
  line-height: 48px;
  max-width: 704px;
  font-weight: 600;
}

.c7 {
  font-size: 14.666666666666666px;
  line-height: 18.666666666666668px;
  max-width: 234.66666666666666px;
  font-weight: 600;
}

.c8 {
  font-size: 17.333333333333332px;
  line-height: 21.333333333333332px;
  max-width: 277.3333333333333px;
  font-weight: 600;
}

.c9 {
  font-size: 22.666666666666664px;
  line-height: 26.666666666666664px;
  max-width: 362.66666666666663px;
  font-weight: 600;
}

.c10 {
  font-size: 28px;
  line-height: 32px;
  max-width: 448px;
  font-weight: 600;
}

.c11 {
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
  font-weight: 600;
}

.c12 {
  font-size: 77px;
  line-height: normal;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

@media only screen and (max-width:512px) {
  .c1 {
    font-size: 17.333333333333332px;
    line-height: 21.333333333333332px;
    max-width: 277.3333333333333px;
  }
}

@media only screen and (max-width:512px) {
  .c2 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c3 {
    font-size: 33.33333333333333px;
    line-height: 37.33333333333333px;
    max-width: 533.3333333333333px;
  }
}

@media only screen and (max-width:512px) {
  .c4 {
    font-size: 44px;
    line-height: 48px;
    max-width: 704px;
  }
}

@media only screen and (max-width:512px) {
  .c5 {
    font-size: 14.666666666666666px;
    line-height: 18.666666666666668px;
    max-width: 234.66666666666666px;
  }
}

@media only screen and (max-width:512px) {
  .c6 {
    font-size: 28px;
    line-height: 32px;
    max-width: 448px;
  }
}

@media only screen and (max-width:512px) {
  .c7 {
    font-size: 12px;
    line-height: 16px;
    max-width: 192px;
  }
}

@media only screen and (max-width:512px) {
  .c8 {
    font-size: 12px;
    line-height: 16px;
    max-width: 192px;
  }
}

@media only screen and (max-width:512px) {
  .c9 {
    font-size: 12px;
    line-height: 16px;
    max-width: 192px;
  }
}

@media only screen and (max-width:512px) {
  .c10 {
    font-size: 12px;
    line-height: 16px;
    max-width: 192px;
  }
}

@media only screen and (max-width:512px) {
  .c11 {
    font-size: 12px;
    line-height: 16px;
    max-width: 192px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
    size="small"
  />
  <h1
    className="c2"
    size="medium"
  />
  <h1
    className="c3"
    size="large"
  />
  <h1
    className="c4"
    size="xlarge"
  />
  <h2
    className="c5"
    size="small"
  />
  <h2
    className="c1"
    size="medium"
  />
  <h2
    className="c2"
    size="large"
  />
  <h2
    className="c6"
    size="xlarge"
  />
  <h3
    className="c7"
    size="small"
  />
  <h3
    className="c8"
    size="medium"
  />
  <h3
    className="c9"
    size="large"
  />
  <h3
    className="c10"
    size="xlarge"
  />
  <h4
    className="c11"
    size="small"
  />
  <h4
    className="c11"
    size="medium"
  />
  <h4
    className="c11"
    size="large"
  />
  <h4
    className="c11"
    size="xlarge"
  />
  <h1
    className="c12"
    size="77px"
  />
</div>
`;

exports[`Heading textAlign renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
  text-align: left;
}

.c2 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
  text-align: center;
}

.c3 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
  text-align: right;
}

@media only screen and (max-width:512px) {
  .c1 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c2 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c3 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <h1
    className="c2"
  />
  <h1
    className="c3"
  />
</div>
`;

exports[`Heading truncate renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

.c2 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

@media only screen and (max-width:512px) {
  .c1 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c2 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    a b c d e f g h i j k l m n o p q r s t u v w x y z
  </h1>
  <h1
    className="c2"
  >
    a b c d e f g h i j k l m n o p q r s t u v w x y z
  </h1>
</div>
`;

exports[`Theme based font family renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-family: Arial;
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

.c2 {
  font-family: Roboto;
  font-size: 22.666666666666664px;
  line-height: 26.666666666666664px;
  max-width: 362.66666666666663px;
  font-weight: 600;
}

.c3 {
  font-family: Ubuntu;
  font-size: 17.333333333333332px;
  line-height: 21.333333333333332px;
  max-width: 277.3333333333333px;
  font-weight: 600;
}

.c4 {
  font-family: Fira Sans;
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
  font-weight: 600;
}

@media only screen and (max-width:512px) {
  .c1 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c2 {
    font-size: 17.333333333333332px;
    line-height: 21.333333333333332px;
    max-width: 277.3333333333333px;
  }
}

@media only screen and (max-width:512px) {
  .c3 {
    font-size: 12px;
    line-height: 16px;
    max-width: 192px;
  }
}

@media only screen and (max-width:512px) {
  .c4 {
    font-size: 12px;
    line-height: 16px;
    max-width: 192px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <h2
    className="c2"
  />
  <h3
    className="c3"
  />
  <h4
    className="c4"
  />
</div>
`;

exports[`Theme based font weight renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 700;
}

.c2 {
  font-size: 22.666666666666664px;
  line-height: 26.666666666666664px;
  max-width: 362.66666666666663px;
  font-weight: 400;
}

.c3 {
  font-size: 17.333333333333332px;
  line-height: 21.333333333333332px;
  max-width: 277.3333333333333px;
  font-weight: 200;
}

.c4 {
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
  font-weight: 600;
}

@media only screen and (max-width:512px) {
  .c1 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

@media only screen and (max-width:512px) {
  .c2 {
    font-size: 17.333333333333332px;
    line-height: 21.333333333333332px;
    max-width: 277.3333333333333px;
  }
}

@media only screen and (max-width:512px) {
  .c3 {
    font-size: 12px;
    line-height: 16px;
    max-width: 192px;
  }
}

@media only screen and (max-width:512px) {
  .c4 {
    font-size: 12px;
    line-height: 16px;
    max-width: 192px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <h2
    className="c2"
  />
  <h3
    className="c3"
  />
  <h4
    className="c4"
  />
</div>
`;

exports[`responsive renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

.c2 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

@media only screen and (max-width:512px) {
  .c1 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <h1
    className="c2"
  />
</div>
`;
